### PR TITLE
[TIMOB-24483] (6_0_X) Updated to ioslib@1.2.0 which improved detection of ent…

### PR DESCRIPTION
**Pull request by cb1kenobi**
Updated to ioslib@1.2.0 which improved detection of enterprise ad hoc provisioning profiles. Then changed the provisioning profile validation to disallow app store pp's for ad hoc builds and ad hoc pp's for app store builds. Added enterprise ad hoc pp's to 'ti info' output.

**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24483

### CODE REVIEW

File: iphone/cli/commands/_build.js [hunk 1]
```diff
@@ -1004,28 +1004,47 @@ iOSBuilder.prototype.configOptionPPuuid = function configOptionPPuuid(order) {
...
-						provisioningProfiles[__('Available Distribution UUIDs:')] = pp;
+						provisioningProfiles[__('Available App Store Distribution UUIDs:')] = pp;
+					} else {
+						logger.error(__('Unable to find any non-expired app store distribution provisioning profiles that match the app id "%s".', appId) + '\n');
+						logger.log(__('You will need to log in to %s with your Apple Developer account, then create, download, and install a profile.',
+							'http://appcelerator.com/ios-dist-certs'.cyan) + '\n');
+						process.exit(1);
 					}
+				} else {
+					logger.error(__('Unable to find any app store distribution provisioning profiles'));
```
Comment (hansemannn) [inline iphone/cli/commands/_build.js]: Name "App Store" as well to match previous usages.

File: iphone/cli/commands/_build.js [hunk 2]
```diff
@@ -1004,28 +1004,47 @@ iOSBuilder.prototype.configOptionPPuuid = function configOptionPPuuid(order) {
...
+					pp = prep(iosInfo.provisioning.enterprise);
 					valid += pp.length;
 					if (pp.length) {
-						provisioningProfiles[__('Available Adhoc UUIDs:')] = pp;
+						provisioningProfiles[__('Available Enterprise Ad Hoc UUIDs:')] = pp;
 					}
 
 					if (!valid) {
-						logger.error(__('Unable to find any non-expired distribution or adhoc provisioning profiles that match the app id "%s".', appId) + '\n');
+						logger.error(__('Unable to find any non-expired ad hoc or enterprise ad hoc provisioning profiles that match the app id "%s".', appId) + '\n');
```
Comment (hansemannn) [inline iphone/cli/commands/_build.js]: "Ad Hoc"

File: iphone/cli/commands/_build.js [hunk 3]
```diff
@@ -1004,28 +1004,47 @@ iOSBuilder.prototype.configOptionPPuuid = function configOptionPPuuid(order) {
...
 						logger.log(__('You will need to log in to %s with your Apple Developer account, then create, download, and install a profile.',
 							'http://appcelerator.com/ios-dist-certs'.cyan) + '\n');
 						process.exit(1);
 					}
 				} else {
-					logger.error(__('Unable to find any distribution or adhoc provisioning profiles'));
+					logger.error(__('Unable to find any ad hoc or enterprise ad hoc provisioning profiles'));
```
Comment (hansemannn) [inline iphone/cli/commands/_build.js]: "Ad Hoc" and "Enterprise" to match previous usages.

File: iphone/cli/lib/info.js [hunk 4]
```diff
@@ -76,7 +76,7 @@ exports.detect = function (types, config, next) {
 					issue.message += '\n' + __('Download and install the certificate from %s', '__http://appcelerator.com/ios-wwdr__');
 					break;
 				case 'IOS_NO_KEYCHAINS_FOUND':
-					issue.message += '\n' + __('Titanium will most likely not be able to detect any developer or distribution certificates.');
+					issue.message += '\n' + __('Titanium will most likely not be able to detect any developer or app store distribution certificates.');
```
Comment (hansemannn) [inline iphone/cli/lib/info.js]: "App Store"

**Review by hansemannn — DISMISSED** [review]
Man, this is great! Left some minor log-style changes to unify the phrases we use. The CR looks cool, the ioslib changes as well. So it was `ProvisionedDevices` vs `ProvisionsAllDevices` that allowed to detect the difference?

**Comment by cb1kenobi** [comment]
Basically, it wasn't much code, but getting there was fun.

**Review by mukherjee2 — APPROVED** [review]
Passed FR using this environment:
NPM Version: 2.15.9
Node Version: 4.5.0
Mac OS: 10.12.3
Appc CLI: 6.1.0
Appc CLI NPM: 4.2.8
Appcelerator Studio, build: 4.8.1.201612050850
Xcode 8.2.1

Passed FR. I created an Adhoc build using the CLI, and no errors were seen.

**Pull request merged**